src/login/checkSettingsOnStartup.ts

```diff
@@ -0,0 +1,84 @@
+
+    // Ask to sign out & reload if the cache is being reset.
+    let shouldSignOutAndReload: boolean = resetLastSeenSettingsCache;
+
+	for (let index = 0; index < numSettings; index++) {
+		const settingKey: string = cachedSettingKeys[index];
+
+		const settingValueOnStartup: string | undefined = getSettingValue(settingKey);
+		sendTelemetryEvent(actionContext, settingKey, 'OnStartup', settingValueOnStartup);
+
+		const lastSeenSettingValue: string | undefined = lastSeenSettingsCacheVerified.values[index];
+		sendTelemetryEvent(actionContext, settingKey, 'LastSeen', lastSeenSettingValue)
+
+        if (settingValueOnStartup !== lastSeenSettingValue) {
+            shouldSignOutAndReload = true;
+        }
+
+		lastSeenSettingsCacheNew.values[index] = settingValueOnStartup;
+	}
+
+	await extensionContext.globalState.update(settingsCacheKey, lastSeenSettingsCacheNew);
+
+	if (shouldSignOutAndReload) {
+		await askThenSignOutAndReload(actionContext, loginHelper);
+    }
+}
+
+export async function askThenSignOutAndReload(actionContext: IActionContext, loginHelper: AzureAccountLoginHelper, forceLogout?: boolean): Promise<void> {
+    actionContext.telemetry.properties.askThenSignOutAndReload = 'true';
+
+    const signOutAndReloadRequired: string = localize('azure-account.signOutAndReloadRequired', 'Signing out and reloading the window is required for the modified setting(s) to take effect.');
+    const signOutAndReload: string = localize('azure-account.signOutAndReload', 'Sign Out and Reload Window');
+    
+    // Purposefully await this message to block whatever command caused the extension to activate.
+    await window.showInformationMessage(signOutAndReloadRequired, signOutAndReload).then(async value => {
+        if (value === signOutAndReload) {
+            actionContext.telemetry.properties.signOutAndReload = 'true';
+            await loginHelper.logout(forceLogout);
+            await commands.executeCommand('workbench.action.reloadWindow');
+        }
+    });
+}
+
+export async function askForSignIn(actionContext: IActionContext): Promise<void> {
+    actionContext.telemetry.properties.askForSignIn = 'true';
+
+    const signInAgainRequired: string = localize('azure-account.signInAgainRequired', 'Signing in again is required for the modified setting(s) to take effect.');
+    const signIn: string = localize('azure-account.signIn', 'Sign In');
+    
+    // Purposefully await this message to block whatever command caused the extension to activate.
+    await window.showInformationMessage(signInAgainRequired, signIn).then(async value => {
+        if (value === signIn) {
+            actionContext.telemetry.properties.signInAgain = 'true';
+            await commands.executeCommand('azure-account.login');
+        }
+    });
+}
+
+function sendTelemetryEvent(actionContext: IActionContext, eventPrefix: string, eventSuffix: string, value: string | undefined): void {
```

Comment:
Maybe this? We aren't actually sending the event in this method.

```suggestion
function addPropertyToTelemetry(actionContext: IActionContext, eventPrefix: string, eventSuffix: string, value: string | undefined): void {
```